warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__shapes_file_present.sql

```diff
@@ -0,0 +1,37 @@
+WITH feed_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__schedule_feed_guideline_index') }}
+),
+
+daily_feed_shapes_present AS (
+    SELECT feed_key,
+           EXTRACT(date FROM ts) AS date,
+           -- zipfile_files is an array which may contain "shapes.txt" but may also contain a string like "google_transit/shapes.txt".
+           -- this long EXISTS statement solves for both
+           CASE WHEN EXISTS(SELECT * FROM UNNEST(zipfile_files) AS x WHERE LOWER(x) LIKE '%shapes.txt') THEN true
```

Comment:
There is also the parse outcomes table: `stg_gtfs_schedule__file_parse_outcomes`, which would record if we had a totally corrupted CSV (and it has the benefit of being already "unnested" so that each individual file has its own row). So I would maybe consider using that instead, so you can check all three outcomes: download, unzip, and parse?

Also, instead of checking `LIKE %shapes.txt` I think it would be safer to just do like `TRIM(LOWER(x)) = 'shapes.txt'` because it is possible that feeds could have wacky other files in them. Unlikely but just in case. And even `TRIM` and `LOWER` gives them the benefit of the doubt if they have spaces or capital letters in their filenames, maybe we don't even want to do that.

Reply:
parse outcomes looks intriguing, I will check it out.
What I was solving for with `LIKE %shapes.txt` is situations where the filename was something like `google_transit/shapes.txt`, which is pretty common in the table I'm using

Reply:
Ah yup I forgot about that -- I definitely think that https://dbt-docs.calitp.org/#!/model/model.calitp_warehouse.stg_gtfs_schedule__file_parse_outcomes is gonna be a better bet, because it has `gtfs_filename` already populated